lib/arithmetic/nodes.rb

```diff
@@ -11,7 +11,17 @@ def to_s(visitor, na=nil)
     end
    
     def eval
-      BigDecimal.new(@operand)
+      if has_dangling_decimal_point?
+        BigDecimal.new(@operand + "0")
+      else
+        BigDecimal.new(@operand)
+      end
+    end
+
+    private
+
+    def has_dangling_decimal_point?
+      @operand.is_a?(String) && @operand[-1] == "."
```

Comment:
I think it' should be `@operand.ends_with?(".")`. Otherwise, something like "13." would still be broken

Reply:
Sorry, I read `-1` as `1`. What you have should be fine then.